File: 2942.find-words-containing-character.java
/*
 * @lc app=leetcode id=2942 lang=java
 *
 * [2942] Find Words Containing Character
 */

// @lc code=start

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> findWordsContaining(String[] words, char x) {
        List<Integer> ret = new ArrayList<>();
        for (int ix = 0; ix != words.length; ++ix) {
            if (words[ix].indexOf(x) != -1) {
                ret.add(ix);
            }
        }
        return ret;
    }
}
// @lc code=end

File: 2942.find-words-containing-character.cpp
/*
 * @lc app=leetcode id=2942 lang=cpp
 *
 * [2942] Find Words Containing Character
 */

// @lc code=start
#include <algorithm>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<int> findWordsContaining(vector<string>& words, char x) {
        vector<int> ret;
        for (auto ix = 0ull; ix != words.size(); ++ix) {
            const auto pos = find(words[ix].cbegin(), words[ix].cend(), x);
            if (pos != words[ix].cend()) ret.push_back(ix);
        }
        return ret;
    }
};
// @lc code=end
